pkg/util/runtime/cpu_linux.go

```diff
@@ -53,6 +60,51 @@ func NumCPU() int {
 	return int(math.Ceil(float64(cpuQuota) / float64(cpuPeriod)))
 }
 
+func getCgroupVersion() int64 {
+	// /sys/fs/cgroup/cgroup.controllers will not exist with cgroupsv1
+	if _, err := os.Stat("/sys/fs/cgroup/cgroup.controllers"); err == nil {
+		return 2
+	} else {
+		return 1
+	}
+}
+
+func readCgroup2FileToInt64Tuple(cgroupFile string) (int64, int64) {
+	contents, err := os.ReadFile(filepath.Join("/sys/fs/cgroup/", cgroupFile))
+
+	if err != nil {
+		return -1, -1
+	}
+
+	// file contents looks like: $MAX $PERIOD
+	// $MAX can have value "max" indicating no limit
+	// it is possible for $PERIOD to be unset
+
+	values := strings.Fields(string(contents))
+
+	if values[0] == "max" {
+		return -1, -1
+	}
+
+	cpuQuota, err := strconv.ParseInt(values[0], 10, 64)
+
+	if err != nil {
+		return -1, -1
+	}
+
+	if len(values) == 1 {
+		return cpuQuota, 1
```

Comment:
From the kernel documentation, the default for $PERIOD is 100000, maybe do the same as default here?

Reply:
Good catch, fixed